src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -75,12 +75,17 @@ internal abstract partial class HttpProtocol : IHttpResponseControl
         private Stream _requestStreamInternal;
         private Stream _responseStreamInternal;
 
-        public HttpProtocol(HttpConnectionContext context)
+        public void Initialize(HttpConnectionContext context)
         {
             _context = context;
 
             ServerOptions = ServiceContext.ServerOptions;
-            HttpRequestHeaders = new HttpRequestHeaders(reuseHeaderValues: !ServerOptions.DisableStringReuse);
+
+            Reset();
+
+            HttpRequestHeaders.ReuseHeaderValues = !ServerOptions.DisableStringReuse;
+            RequestHeaders = RequestHeaders;
```

Comment:
This line is a bit weird?

Reply:
Yeah, I don't need that anymore. Before, this was newing up HttpRequestHeaders rather than changing a property.

Reply:
And it was supposed to be `RequestHeaders = HttpRequestHeaders;`